src/RSocketClient.cpp

```diff
@@ -41,11 +44,11 @@ folly::Future<std::unique_ptr<RSocketRequester>> RSocketClient::connect(
     keepaliveTimer = std::move(keepaliveTimer),
     stats = std::move(stats),
     networkStats = std::move(networkStats),
-    promise = std::move(promise)](
-      std::unique_ptr<DuplexConnection> connection,
-      folly::EventBase& eventBase) mutable {
+    promise = std::move(promise)
+  ](std::unique_ptr<DuplexConnection> connection,
+    folly::EventBase & eventBase) mutable {
     VLOG(3) << "onConnect received DuplexConnection";
-
+    evb_ = &eventBase;
```

Comment:
The "Scheduled" wrappers are for Subscriber/Subscription/Responder, not for stateMachine/duplexConnection.  Here the client schedules a disconnect() directly with the stateMachine, so I have used a raw-evb.  

The check in resumption is for ensuring that the resumed connection is also running in the same evb as the old connection.   With TcpConnectionFactory, this check is unnecessary as the subsequent clients will be in the same worker_.evb.  But a custom ConnectionFactory could give out clients in different evbs for every connect().  At that point we dont want to run our stateMachine bound to old evb_ in this new evb_.  Ondrej has a task to add ability to migrate stateMachines across evbs.  That could change what we do here.